msteams-platform/toolkit/configure-message-extension-capability.md

```diff
@@ -0,0 +1,632 @@
+                "type": "query",
+                "parameters": [
+                    {
+                        "name": "searchQuery",
+                        "title": "Search Query",
+                        "description": "Your search query",
+                        "inputType": "text"
+                    }
+                ]
+            }
+        ],
+        "messageHandlers": [
+            {
+                "type": "link",
+                "value": {
+                    "domains": [
+                        "*.botframework.com"
+                    ]
+                }
+            }
+        ]
+    }
+]
+```
+
+### Bring message extension code to your project
+
+1. You can bring your message extension app code into your project. If you don't have an app code, use the previously created message extension app project and copy the source code into your current project. We suggest you copy them into a `bot/` folder. The following folder structure is an example of how your project's folder structure looks like:
+
+   |--.vscode/
+   |--appPackage/
+  |--env/
+   |--infra/
+   |--public/
+   |--bot/           <!--message extension source code-->
+   |   |--index.ts
+   |   |--config.ts
+   |   |--teamsBot.ts
+   |   |--package.json
+   |   |--tsconfig.json
+   |   |--web.config
+   |   |--.Webappignore
+   |--src/            <!--your current source code-->
+   |   |--app.ts
+   |   |--static/
+   |   |--views/
+   |--package.json
+   |--tsconfig.json
+   |--teamsapp.local.yml
+   |--teamsapp.yml
+
+   You can create a root `package.json` using command `npm init -y`. We recommend you reorganize the folder structure and create a root `package.json` as given in the following folder structure:
+
+   |--.vscode/
+   |--appPackage/
+   |--env/
+   |--infra/
+   |--bot/            <!--message extension source code-->
+      |--index.ts
+   |   |--config.ts
+   |   |--teamsBot.ts
+   |   |--package.json
+   |   |--tsconfig.json
+   |   |--web.config
+   |   |--.Webappignore
+   |--tab/           <!--move your current source code to a new sub folder-->
+   |   |--src/
+   |   |   |--app.ts
+   |   |   |--static/
+   |   |   |--views/
+   |   |--package.json
+   |   |--tsconfig.json
+   |--package.json <!--root package.json-->
+   |--teamsapp.local.yml
+   |--teamsapp.yml
+
+1. Add the following code to your root `package.json`:
+
+   ```json
+   "scripts": {
+     "test": "echo \"Error: no test specified\" && exit 1",
+     "install:bot": "cd bot && npm install",
+     "install:tab": "cd tab && npm install",
+     "install": "concurrently \"npm run install:bot\" \"npm run install:tab\"",
+     "dev:bot": "cd bot && npm run dev",
+     "start:tab": "cd tab && npm run start",
+     "build:tab": "cd tab && npm run build",
+     "build:bot": "cd bot && npm run build",
+     "build": "concurrently \"npm run build:tab\" \"npm run build:bot\""
+   },
+   "dependencies": {
+       "concurrently": "^7.6.0"
+   },
+   ```
+
+> [!NOTE]
```

Comment:
Give tab space and align it below the code space

Reply:
done